src/OAuth2Form.tsx

```diff
@@ -1,403 +1,439 @@
         })
-        .catch((_err) => {
-          onError(_err);
-        });
+        .catch(_err => {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+          onError(_err)
+        })
     }
     return () => {
-      isSubscribed = false;
-    };
-  }, [accessToken, isConsenting, onError]);
+      isSubscribed = false
+    }
+  }, [accessToken, isConsenting, onError])
 
   const onGoBack = () => {
-    window.history.back();
-  };
+    window.history.back()
+  }
 
   const onDeny = () => {
-    let redirect: string;
-    sendGTagEvent("UserDeniedConsent");
+    let redirect: string
+    sendGTagEvent('UserDeniedConsent')
     if (oauthClientInfo && oauthClientInfo.client_uri) {
-      redirect = oauthClientInfo.client_uri;
+      redirect = oauthClientInfo.client_uri
     } else {
-      redirect = getURLParam("redirect_uri")!;
+      redirect = getURLParam('redirect_uri')!
     }
-    setRedirectURL(redirect);
-  };
+    setRedirectURL(redirect)
+  }
 
   // if user has already consented to this client request, no need to ask again
   useEffect(() => {
     const getHasAlreadyConsented = () => {
-      const code = getURLParam("code");
-      if (code) return; // we're in the middle of a SSO, do not attempt to get OAuth2RequestDescription yet
+      const code = getURLParam('code')
+      if (code) return // we're in the middle of a SSO, do not attempt to get OAuth2RequestDescription yet
       const request: OIDCAuthorizationRequest =
-        getOIDCAuthorizationRequestFromSearchParams();
+        getOIDCAuthorizationRequestFromSearchParams()
       SynapseClient.hasUserAuthorizedOAuthClient(request, accessToken!)
         .then((consentGrantedResponse: OAuthConsentGrantedResponse) => {
-          const prompt = getURLParam("prompt");
+          const prompt = getURLParam('prompt')
           if (consentGrantedResponse.granted) {
             // SWC-5285: before auto-consenting, make sure we're allowed to auto-consent.
             // Only allow if prompt is undefined or set to none.
-            if (!prompt || prompt !== "consent") {
+            if (!prompt || prompt !== 'consent') {
               // auto-consent!
-              onConsent();
+              onConsent()
             } //else if prompt is defined and another value ('login', 'consent', or 'select_account') then always prompt!
-          } else if (prompt && prompt === "none") {
+          } else if (prompt && prompt === 'none') {
             // granted === false and prompt === none
             onError(
               new OAuthClientError(
-                "consent_required",
-                "Current user has not previously granted permission, and prompt was set to none"
-              )
-            );
+                'consent_required',
+                'Current user has not previously granted permission, and prompt was set to none',
+              ),
+            )
           }
+          setIsPreviousAuthCheckComplete(true)
```

Comment:
When the user profile is available (user is logged in), hasUserAuthorizedOAuthClient() is called and isPreviousAuthCheckComplete is set to true